File: 2501-3000/2526.Find Consecutive Integers from a Data Stream.rb
class DataStream

=begin
    :type value: Integer
    :type k: Integer
=end
    def initialize(value, k)
        @value = value
        @k = k
        @acc_cnt = 0
    end


=begin
    :type num: Integer
    :rtype: Boolean
=end
    def consec(num)
        @acc_cnt = num == @value ? @acc_cnt + 1 : 0
        return @acc_cnt >= @k
    end


end

# Your DataStream object will be instantiated and called as such:
# obj = DataStream.new(value, k)
# param_1 = obj.consec(num)

File: 2501-3000/2526.Find Consecutive Integers from a Data Stream.php
class DataStream {

    private $value;
    private $k;
    private $cnt;

    /**
     * @param Integer $value
     * @param Integer $k
     */
    function __construct($value, $k) {
        $this->value = $value;
        $this->k = $k;
        $this->cnt = 0;

    }

    /**
     * @param Integer $num
     * @return Boolean
     */
    function consec($num) {
        $this->cnt = ($num == $this->value) ? $this->cnt + 1 : 0;
        return $this->cnt >= $this->k;
    }
}

/**
 * Your DataStream object will be instantiated and called as such:
 * $obj = DataStream($value, $k);
 * $ret_1 = $obj->consec($num);
 */

